internal/stream_fetch_helper.go

```diff
@@ -132,11 +111,66 @@ func DownloadAndDecompressSplittedStream(backup Backup, blockSize int, extension
 	return lastErr
 }
 
-func detectPartitionsCount(backup Backup, decompressor compression.Decompressor) (int, error) {
+func downloadFile(backup Backup, decompressor compression.Decompressor, fileName string, writer io.WriteCloser) error {
```

Comment:
We have added 4 extra functions.

May we just modify `detectPartitionsCount` to return `[]int` (number of part for each partition).
And add one loop (over parts) in `DownloadAndDecompressSplittedStream` ?